### PR TITLE
Feature/rc retry

### PR DESCRIPTION
**Pull request by rodrigopavezi**
@donnapep @stulees please review.

You will need to run RC locally when previewing this presentation http://rva.risevision.com/#/PRESENTATION_MANAGE/id=914abe63-6d95-4485-9012-d9f85b0445dd?cid=30007b45-3df0-4c7b-9f7f-7d8ce6443013

cheers

### CODE REVIEW

File: src/widget/non-storage.js [hunk 1]
```diff
@@ -33,26 +33,34 @@ RiseVision.Video.NonStorage = function( data ) {
...
-        RiseVision.Video.logEvent( {
-          "event": "non-storage error",
-          "event_details": error.message,
-          "file_url": response.url
-        }, true );
-
-        if ( riseCache.isV2Running() ) {
-          errorMessage = riseCache.getErrorMessage( statusCode );
+
+        if ( error.message == "File is downloading" ) {
```
Comment (stulees) [inline src/widget/non-storage.js]: @rodrigopavezi Should this check that `error.message` exists before checking for value?

File: test/integration/non-storage/messaging.html [hunk 2]
```diff
@@ -202,9 +206,18 @@
         clock.tick(500);
         assert(spy.calledOnce);
 
-        clock.restore();
         RiseVision.Video.play.restore();
       });
+
+      test("should show File is downloaidng message when RC responds with a 202 for 3 consecutive times", function() {
+        requests[1].respond(202); // HEAD request
```
Comment (donnapep) [inline test/integration/non-storage/messaging.html]: @rodrigopavezi Same comment as Image. How do we know this is happening after 3 tries? The message could have shown up after the first tick. Perhaps asserting that the message is still empty after each tick?

File: test/integration/non-storage/messaging.html [hunk 3]
```diff
@@ -202,9 +206,18 @@
...
+        clock.tick(3000);
+        requests[2].respond(202); // HEAD request
+        clock.tick(3000);
+        requests[3].respond(202); // HEAD request
+        assert.equal(document.querySelector(".message").innerHTML, "File is downloading", "file is downloading message text");
```
Comment (donnapep) [inline test/integration/non-storage/messaging.html]: @rodrigopavezi Same comment as Image. How do we know this is happening after 3 tries? The message could have shown up after the first tick. Perhaps asserting that the message is still empty after each tick?

File: src/widget/non-storage.js (hunk 1 above)
Comment (rodrigopavezi) [inline src/widget/non-storage.js]: @stulees I think the check will resolve to undefined == "File is downloading" which should work without exception. I think what you are woried is in the case the error object is undefined but that is not the case here. I haven't test that I might be wrong. I made the change anyway to be safe. cheers

**Comment by rodrigopavezi** [comment]
@donnapep @stulees changes made. please review. cheers

File: test/integration/non-storage/messaging.html [hunk 4]
```diff
@@ -212,8 +212,10 @@
       test("should show File is downloaidng message when RC responds with a 202 for 3 consecutive times", function() {
         requests[1].respond(202); // HEAD request
         clock.tick(3000);
+        assert.isFalse((document.getElementById("messageContainer").style.display === "node"), "message visibility");
```
Comment (donnapep) [inline test/integration/non-storage/messaging.html]: @rodrigopavezi I think this should be "none" and it should be `true`. Same thing for the other one.

File: src/widget/non-storage.js [hunk 5]
```diff
@@ -34,7 +34,7 @@ RiseVision.Video.NonStorage = function( data ) {
 
       } else {
 
-        if ( error.message == "File is downloading" ) {
+        if ( error.message && error.message == "File is downloading" ) {
```
Comment (donnapep) [inline src/widget/non-storage.js]: @rodrigopavezi Could this be changed to ===?

Comment (rodrigopavezi) [inline src/widget/non-storage.js]: @donnapep yeap, changed. cheers

**Comment by donnapep** [comment]
👍

**Pull request merged**